Code/GraphMol/PeriodicTable.cpp

```diff
@@ -79,9 +83,18 @@ PeriodicTable::PeriodicTable() {
   }
 }
 
+#if RDK_BUILD_THREADSAFE_SSS
+boost::mutex ptMutex;
+#endif
+
 PeriodicTable *PeriodicTable::getTable() {
   if (ds_instance == 0) {
-    ds_instance = new PeriodicTable();
+#if RDK_BUILD_THREADSAFE_SSS
+    boost::lock_guard<boost::mutex> lock(ptMutex);
```

Comment:
In general it is faster to use boost::call_once as this doesn't have to use a mutex on every call to getTable.   (and double checked locking is broken unless using c++11).  If this is used often, it may be useful to profile this to see the impact of the mutex contention.

Interestingly, gcc and clang have thread safe initialization of statics but MSVC doesn't which is why we need this nonsense in the first place, so we may be able to optimize this for linux/osx...

It does trouble me a bit returning a non-const object to multiple threads though, how often is this modified?